ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/ReportJobStatus.tsx

```diff
@@ -0,0 +1,95 @@
+const disabledColor = 'var(--pf-global--disabled-color--100)';
+
+const genericMsg =
+    'An issue was encountered. Please try again later. If the issue persists, please contact support';
+
+function ReportJobStatus({ reportSnapshot }: ReportJobStatusProps): ReactElement {
+    const { reportStatus, isDownloadAvailable } = reportSnapshot;
+
+    let statusIcon: ReactElement;
+    let statusText: ReactElement;
+
+    if (
+        reportStatus.runState === 'SUCCESS' &&
+        reportStatus.reportNotificationMethod === 'DOWNLOAD' &&
+        isDownloadAvailable
+    ) {
+        statusIcon = (
+            <DownloadIcon color={infoColor} title="Report download was successfully prepared" />
+        );
+        statusText = <p className="pf-u-info-color-100">Ready for download</p>;
+    } else if (
+        reportStatus.runState === 'SUCCESS' &&
+        reportStatus.reportNotificationMethod === 'DOWNLOAD' &&
+        !isDownloadAvailable
+    ) {
+        statusIcon = <DownloadIcon color={disabledColor} title="Report download was deleted" />;
+        statusText = (
+            <Flex
+                direction={{ default: 'row' }}
+                spaceItems={{ default: 'spaceItemsSm' }}
+                alignItems={{ default: 'alignItemsCenter' }}
+            >
+                <FlexItem>
+                    <p className="pf-u-disabled-color-100">Ready for download</p>
+                </FlexItem>
+                <FlexItem>
+                    <Tooltip
+                        content={
+                            <div>
+                                The download was deleted. Please generate a new download if needed.
+                            </div>
+                        }
+                    >
+                        <HelpIcon className="pf-u-disabled-color-100" />
+                    </Tooltip>
+                </FlexItem>
+            </Flex>
+        );
+    } else if (reportStatus.runState === 'SUCCESS') {
+        statusIcon = <CheckCircleIcon color={successColor} title="Report run was successful" />;
+        statusText = <p className="pf-u-success-color-100">Successful</p>;
+    } else if (reportStatus.runState === 'FAILURE') {
+        statusIcon = (
+            <Tooltip content={reportStatus?.errorMsg || genericMsg}>
+                <ExclamationCircleIcon color={errorColor} title="Report run was unsuccessful" />
+            </Tooltip>
+        );
+        statusText = <p className="pf-u-danger-color-100">Error</p>;
+    } else if (reportStatus.runState === 'PREPARING') {
+        statusIcon = <InProgressIcon title="Report run is preparing" />;
+        statusText = <p>Preparing</p>;
+    } else {
+        statusIcon = <PendingIcon title="Report run is waiting" />;
+        statusText = <p>Waiting</p>;
+    }
+
+    return (
+        <Flex alignItems={{ default: 'alignItemsCenter' }}>
+            {statusIcon && <FlexItem>{statusIcon}</FlexItem>}
+            <FlexItem>{statusText}</FlexItem>
+        </Flex>
```

Comment:
Small nit: It looks like in each case where the color is different from the default, the text and icon are both the same color. You may be able to set a `color` variable in each block where this is that case and `style={{ color }}` on the outer `<Flex>` here to have all child elements inherit it, which would cut down some of the repetition and remove the need to use both PF utility classes and CSS variables.